### ISSUE TITLE
multi_match cross_fields type does not work with position_offset_gap setting (!?)

### ISSUE THREAD

**Issue by dezsenyi** [elastic/elasticsearch#9862]
Hi,

It seems that the __cross_fields__ type __multi_match__ query cannot work together with fields that have __position_offset_gap__ settings.
Is this a feature or bug?

Reproduction:

```
# Delete old index
curl -XDELETE "http://localhost:9200/testindex"

# Create test index
curl -XPOST "http://localhost:9200/testindex/" -d'
{
  "index": {
    "number_of_replicas": "0",
    "number_of_shards": "1"
  }
}'

# (testcase 1) Create mapping with position offset gap
curl -XPUT "http://localhost:9200/testindex/_mapping/user" -d'
{
  "user": {
    "properties": {
      "first_name": {
        "type": "string",
        "position_offset_gap": 100
      },
      "last_name": {
        "type": "string",
        "position_offset_gap": 100
      }
    }
  }
}'

# (testcase 2) Create mapping without position offset gap
curl -XPUT "http://localhost:9200/testindex/_mapping/user" -d'
{
  "user": {
    "properties": {
      "first_name": {
        "type": "string"
      },
      "last_name": {
        "type": "string"
      }
    }
  }
}'

# Put 1 user
curl -XPUT "http://localhost:9200/testindex/user/1" -d'
{
  "first_name": "John",
  "last_name": "Smith"
}'

# Multi match with cross_fields (try with testcase 1 and 2)
curl -XGET "http://localhost:9200/testindex/user/_search" -d'
{
  "query": {
    "multi_match": {
      "query": "John Smith",
      "type": "cross_fields",
      "fields": [
        "first_name",
        "last_name"
      ],
      "operator": "and"
    }
  }
}'
```

Try it with the first and then with the second mapping, as well. 
Testcase 1 will not produce result hit, while testcase 2 will get the 1 document.
I tried with the latest ES version.

Thanks,
Csaba D.

**Comment by clintongormley** [comment]
Nice catch!  Confirmed that this is a bug in 1.4.4

thanks

**Comment by VicsCaba** [comment]
Hi,

Has this issue been solved? I was experiencing also some problems with position_offset_gap when applying analyzers. Field is multivalue and position_offset_gap works properly if I don't define any analyzer for this field. However when applying an analyzer such us asciifolding it seems that position_offset_gap is not applied. Is this the expected behaviour?

Thank you

**Comment by clintongormley** [comment]
Hi @VicsCaba - not fixed yet, still on the list.

**Comment by clintongormley** [comment]
This appears to have been fixed, at least in 2.1.0
